packages/jsii-rosetta/lib/jsii/assemblies.ts

```diff
@@ -257,3 +265,19 @@ function findPackageJsonLocation(currentPath: string): string | undefined {
     currentPath = parentPath;
   }
 }
+
+/**
+ * Find the jsii [sub]module that contains the given FQN
+ *
+ * Return `undefined` if the type is a member of the assembly root.
+ */
+export function findContainingSubmodule(assembly: spec.Assembly, fqn: string): string | undefined {
+  const submoduleNames = Object.keys(assembly.submodules ?? {});
+  sortBy(submoduleNames, (s) => [-s.length]); // Longest first
+  for (const s of submoduleNames) {
+    if (fqn.startsWith(`${s}.`)) {
+      return s;
+    }
+  }
+  return undefined;
```

Comment:
Very much optional, but this can be done in a purely functional style (I'm not sure either form is clearly more readable than the other).

```suggestion
  Object.keys(assembly.submodules ?? {})
    // Keep only those that are prefixes of our fqn
    .filter((s) => fqn.startsWith(`${s}.`))
    // Sort by length (longest to shortest)
    .sort((l, r) => r.length - l.length)
    // Longest prefix is the one we're looking for
    [0]
```

Reply:
Nice.